Utils/Dataflow/019_esFormat/esFormat.php

```diff
@@ -5,14 +5,60 @@ function exception_error_handler($errno, $errstr, $errfile, $errline ) {
 }
 set_error_handler("exception_error_handler");
 
+# Default values.
+# ES index where the documents should be indexed/updated.
 $DEFAULT_INDEX = 'tasks_production';
+# End-of-process marker, depending on mode.
+$EOP_DEFAULTS = Array("stream" => chr(0), "file" => "");
+# End-of-message marker, depending on mode.
+$EOM_DEFAULTS = Array("stream" => chr(30), "file" => chr(30));
+
+# Variable initialization.
+# Action. Possible values are 'index' and 'update'. Note: this variable is not
+# a constant and can be changed with command line argument. The "DEFAULT" in
+# its name refers to action not being specified in the data being processed,
+# not in the script's configuration (like the default values above).
 $DEFAULT_ACTION = 'index';
```

Comment:
> The `$DEFAULT_ACTION` description is not very clear to me.
> 
> First of all, I'd recommend to start every section of the description with a new line:
> 
> ```
> # Action
> # Possible values: ...
> # Note: ...
> # ...
> ```

Looks reasonable to me, will do.

> Then, what is unclear to me:
> 
> > this variable is not a constant...
> 
> Pretty obvious: variables usually are _not_ constants. What does it mean?
> 
> > ...and can be changed with command line argument.
> 
> Er... I mean right, it can be changed, but is it important at this point?
> 
> > action not being specified in the data being processed, not in the script's configuration (like the default values above).
> 
> Too many "not"s: while reading, I tend to be treat them as more or less similar "not"s, but they are not... and even though I know it for now, I totally fail to read this phrase properly.
> 
> But before we start looking for better wording, I have a question: is it necessary to add this note here? I mean, whatever global variable it is, here all that you want is to understand its meaning; how it is used and where its value is altered is the second question, which is more likely to be investigated by looking into the code itself, not via reading the explaination of all occurences of the variable name gathered in one place... since to understand the latter you still need to know something about the code.

All these concerns make sense, and can be resolved by changing the name of the variable. The name - or, more precisely, the presence of "DEFAULT" in it - is what made me write this lengthy comment.  There are 6 variables initialized here, ones from the first group are used as constants (their values are set here, and not changed later), and the other ones are used normally. Only "constants" have the word "DEFAULT" in their names, hence I decided to point out that `$DEFAULT_ACTION` is different.

Actually, it seems that I was wrong in the groups' construction, because `$UPDATE_RETRIES` is also a "constant". Therefore, such division of "DEFAULT/no DEFAULT" makes even less sense. Maybe the comment can be shortened without renaming the variable...

Reply:
Things may be unified even better, if we introduce another variable `$ACTION`:
```
$DEFAULT_ACTION = 'index'

<...>

function getAction($row) {
  global $ACTION;
  if (isset($row['_update']) and $row['_update'] === true) {
    $action = 'update';
  } else {
    $action = $ACTION;
  }
  return $action;
}

<...>
    case "update":
      $ACTION = "update";
      break;
<...>

if (!isset($ACTION)) $ACTION = $DEFAULT_ACTION;
```

...and remove `$ES_INDEX` initialization with `null` from top of the file -- anyway it is set to the env variable value or `$DEFAULT_INDEX` later, when all other variables like `$EOM_MARKER` get their values.

Reply:
Done.